### PR TITLE
Fix PSR container conflicts

### LINKED ISSUE #7488: Fatal error: Declaration of WCPay\Internal\DependencyManagement\DelegateContainer\LegacyContainer::has($id) must be compatible with Psr\Container\ContainerInterface::has(string $id): bool
With PHP 8.2.4 I got error

```
 Fatal error: Declaration of WCPay\Internal\DependencyManagement\DelegateContainer\LegacyContainer::has($id) must be compatible with Psr\Container\ContainerInterface::has(string $id): bool
```
In 3 files:
- woocommerce-payments/src/Container.php
- woocommerce-payments/src/Internal/DependencyManagement/DelegateContainer/LegacyContainer.php
- woocommerce-payments/src/Internal/DependencyManagement/DelegateContainer/WooContainer.php

### PR DESCRIPTION
**Pull request by RadoslavGeorgiev**
Fixes #7488

#### Changes proposed in this Pull Request

Previously https://github.com/Automattic/woocommerce-payments/pull/7514, but it was reverted.

#### Testing instructions

Overall, if the container package is in our namespace, and a simple checkout flow works, we should be good.

Specific Testing Instructions for #7488 

1. On current stable release, install and activate [backwpup](https://wordpress.org/plugins/backwpup/) plugin. Observe Payment dashboard screens are not accessible.
2. Update with the release zip (see Slack) **manually**. There should be no error messages or warnings.
3. With this fix, check that Payment dashboards screens work, checkout and payment admin flows are functional

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7525` or branch name `fix/psr-container-conflicts` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/psr-container-conflicts) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 8bbfb1bf24a9552899575ff91e8e4fb9f0b4655b
- Build time: 2023-10-20 10:31:36 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: src/Container.php [hunk 1]
```diff
@@ -16,6 +16,25 @@
...
+ * the ContainerInterface would still be in a different namespace, and would not exist here.
+ *
+ * Preventing the class from being loaded here does nothing but hide the error.
+ * All later requests will work properly.
+ */
+if (
+	! interface_exists( ContainerInterface::class )
+	&& isset( $_GET['action'] ) // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+	&& 'upload-plugin' === $_GET['action'] // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+	&& 'update.php' === $GLOBALS['pagenow']
```
Comment (htdat) [inline src/Container.php:34]: May check `isset` for `pagenow` too?

Comment (RadoslavGeorgiev) [reply src/Container.php:34]: Done in https://github.com/Automattic/woocommerce-payments/pull/7525/commits/c6d4d49d870f15750036c29d32a8ac64db561890

**Review by htdat — APPROVED** [review]
✅ Reviewed code, and manually tested. 

Ref p1697797515711159/1697705563.627949-slack-C01BPL3ALGP

**Pull request merged**